File: src/main/java/com/amigos/productsize/ProductSizeService.java
package com.amigos.productsize;

import com.amigos.common.ResponseApi;
import com.amigos.dto.ProductSizeDTO;

import java.util.UUID;

public interface ProductSizeService
{
    ResponseApi addProductSize(ProductSizeDTO productSizeDTO);
    ResponseApi getDetail(UUID id);
    ResponseApi delete(UUID id);
    ResponseApi getAllByStatus(boolean status);
    ResponseApi getAllProductSizeByCateId(UUID cateId);
    ResponseApi findProductSizeByProductId(UUID productId);
}

File: src/app/authentication/entity/product-size/product-size.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ResponseApi } from '../ResponseApi';
import { ProductSize } from './product-size';

@Injectable({
  providedIn: 'root'
})
export class ProductSizeService {
  private baseUrl = "http://localhost:8081/product-size";

  constructor(private httpClient: HttpClient) { }

  addProductSize(file: any): Observable<Object> {
    return this.httpClient.post(`${this.baseUrl}`, file);
  }

  getAllByStatus(status:any): Observable<ResponseApi>  {
    return this.httpClient.get<ResponseApi>(`${this.baseUrl+"/status"}/${status}`);
  }

  getDetail(id: number): Observable<ResponseApi> {
    return this.httpClient.get<ResponseApi>(`${this.baseUrl}/${id}`);
  }

  updateProduct(file: any): Observable<Object> {
    return this.httpClient.patch(`${this.baseUrl}`, file);
  }

  updateById(productSize: ProductSize): Observable<Object> {
    return this.httpClient.patch(`${this.baseUrl}`, productSize);
  }

  deleteById(id: number): Observable<Object> {
    return this.httpClient.delete(`${this.baseUrl}/${id}`);
  }
}
